slides/aml-11-022118-gradient-boosting-calibration/index.html

```diff
@@ -0,0 +1,646 @@
+          float: right;
+          width: 32%
+      }
+      .wide-left-column {
+          float: left;
+          width: 65%
+      }
+      .invisible {
+          visibility: hidden
+      }
+      .remark-code, .remark-inline-code { font-family: 'Ubuntu Mono'; }
+      .column:first-of-type {float:left}
+      .column:last-of-type {float:right}
+      .remark-code, .remark-inline-code .tiny-code{
+        font-size: 15px;
+      }
+      .split-40 .column:first-of-type {width: 50%}
+      .split-40 .column:last-of-type {width: 50%}
+    </style>
+  </head>
+  <body>
+    <textarea id="source">
+
+class: center, middle
+
+### W4995 Applied Machine Learning
+
+# Boosting, Stacking, Calibration
+
+02/21/18
+
+Andreas C. Müller
+
+???
+N/A
+
+---
+class: centre,middle
+# Gradient Boosting
+???
+---
+
+# Gradient Boosting Algorithm
+
+![:scale 10%](images/img_1.png) ![:scale 20%](images/img_2.png) ![:scale 10%](images/img_8.png) ![:scale 20%](images/img_3.png) ![:scale 10%](images/img_8.png) ![:scale 20%](images/img_4.png)
+
+
+.center[
+![:scale 20%](images/img_5.png)]
+
+.center[
+![:scale 30%](images/img_6.png)
+]
+.center[
+![:scale 30%](images/img_7.png)
+]
+
+- Iteratively add regression trees to model
+- Use log loss for classification
+- Discount update by learning rate
+???
+Come back to this
+---
+#GradientBoostingClassifier
+
+.center[
+![:scale 70%](images/img_9.png)
+]
+???
+
+---
+class:spacious
+# Gradient Boosting
+ - Many shallow trees
+ - learning_rate ↔ n_estimators
+ - Slower to train than RF (serial), but much faster to predict
+ - Small model size
+ - Uses one-vs-rest for multi-class!
+???
+---
+
+class:spacious
+# Tuning Gradient Boosting
+- Pick n_estimators, tune learning rate
+- Can also tune max_features
+- Typically strong pruning via max_depth
+
+???
+---
+
+# Partial Dependence Plots
+- Marginal dependence of prediction on one or two features
+.smaller[
+```python
+boston = load_boston()
+X_train, X_test, y_train, y_test = 
+train_test_split(boston.data, boston.target,random_state=0)
+gbrt = GradientBoostingRegressor().fit(X_train, y_train)
+gbrt.score(X_test, y_test)
+from sklearn.ensemble.partial_dependence import plot_partial_dependence
+fig, axs = plot_partial_dependence(gbrt, X_train, np.argsort(gbrt.feature_import
+                                      ances_)[-6:],feature_names=boston.feature_name
+                                      s,n_jobs=3, grid_resolution=50)
+plt.tight_layout()
+```
+```0.81833662390886375```
+]
+
+???
+---
+.center[
+Plot depicting importance of features vs Partial Dependence
+![:scale 70%](images/img_10.png)
+]
+
+???
+---
+.smaller[
+```python
+fig, axs = plot_partial_dependence(gbrt, X_train, [np.argsort(gbrt.feature_importances_)[-2:]],
+                                   feature_names=boston.feature_names,
+                                   n_jobs=3, grid_resolution=50)
+```
+]
+.center[
+![:scale 70%](images/img_11.png)
+]
+.center[
+Insert Caption
```

Comment:
It's not so much about having a caption. I like to keep the slides clean and provide either video or presenter notes. The idea was to have filenames that are not ``img_11.png`` but rather ``partial_dependence_plot.png``.

Reply:
Oh I see what you mean. I'll change that